tests.cpp

```diff
@@ -1,26 +1,162 @@
 #include "libssh2.hpp"
 #include <iostream>
 #include <bitset>
+#ifdef WIN32
+#include <windows.h>
+#else
+#include <termios.h>
+#include <unistd.h>
+#endif
 
-int main(int argc, char **argv) {
-    std::cout << "Hello, world!" << std::endl;
+void SetStdinEcho(bool enable = true)
+{
+#ifdef WIN32
+    HANDLE hStdin = GetStdHandle(STD_INPUT_HANDLE);
+    DWORD mode;
+    GetConsoleMode(hStdin, &mode);
+
+    if (!enable)
+        mode &= ~ENABLE_ECHO_INPUT;
+    else
+        mode |= ENABLE_ECHO_INPUT;
+
+    SetConsoleMode(hStdin, mode);
+
+#else
+    struct termios tty;
+    tcgetattr(STDIN_FILENO, &tty);
+    if (!enable)
+        tty.c_lflag &= ~ECHO;
+    else
+        tty.c_lflag |= ECHO;
+
+    (void)tcsetattr(STDIN_FILENO, TCSANOW, &tty);
+#endif
+}
+
+int main(int argc, char **argv)
+{
+
+    std::string username, password;
+    size_t nread;
+    char *ptr;
+    char mem[1024];
+    int rc;
+
+    std::cout << "SCP Test" << std::endl;
+    
+    libssh2::session s;
     try
     {
-        libssh2::session s;
-        s.open("127.0.0.1", 8989);
+
+        s.open("127.0.0.1", 22);
         libssh2::fingerprint f = s.get_host_fingerprint();
         std::cout << "SHA1 fingerprint: " << f.get_hex_sha1() << std::endl
-            << "MD5  fingerprint: " << f.get_hex_md5() << std::endl;
-            
-        std::cout << "Auth methods for ruben: " << std::bitset<8>(s.get_auth_methods("ruben")) << std::endl;
-        
-        s.auth_password("username","password");
-        libssh2::channel* c = s.open_channel();
-        c->request_pty();
+                  << "MD5  fingerprint: " << f.get_hex_md5() << std::endl;
+
+        std::cout << " Enter Username: ";
+        std::cin >> username;
+
+        std::cout << " Password: ";
+        SetStdinEcho(false);
+        std::cin >> password;
+        SetStdinEcho(true);
+
+        FILE *local = fopen("./test.txt", "rb");
+            if(!local) {
+                std::cerr << "Can't open local file " << "./test.txt" << std::endl;
+            }
+
+        std::cout << "Auth methods for username: " << username << " " << std::bitset<8>(s.get_auth_methods(username)) << std::endl;
+
+        s.auth_password(username, password);
+        libssh2::channel *c = s.open_scp_channel("/home/markus/test.txt", local);
+        //c->request_pty();
+
+        if (!c)
+        {
+            char *errmsg;
+            int errlen;
+            int err = libssh2_session_last_error(s.get_session_ptr(), &errmsg, &errlen, 0);
+
+            fprintf(stderr, "Unable to open a session: (%d) %s\n", err, errmsg);
+            goto shutdown;
```

Comment:
Please don't use `goto` or `fprintf`, C++ has better features than that. Use destructors for shutdown and `cout` and `cerr`.

Reply:
Sorry, please forget that pull request, I mixed it up with some tests from my side, I will resend an Request when I have time.